tools/scripts/audit-streamhub.sh

```
@@ -171,13 +171,13 @@ for test_file in "${test_files[@]}"; do
         has_quote_observer_method=$(grep -c "QuoteObserver_WithWarmupLateArrivalAndRemoval_MatchesSeriesExactly" "$test_file" || true)
 
         if [[ $has_quote_observer_method -gt 0 ]]; then
-            # Check if the method includes Insert and Remove operations
-            has_insert=$(grep -A 50 "QuoteObserver_WithWarmupLateArrivalAndRemoval_MatchesSeriesExactly" "$test_file" | grep -c "\.Insert(" || true)
+            # Check if the method includes late Add and Remove operations
+            has_add=$(grep -A 50 "QuoteObserver_WithWarmupLateArrivalAndRemoval_MatchesSeriesExactly" "$test_file" | grep -E -c "\\.Add\\([^)]*\\[80\\]\\)" || true)
             has_remove=$(grep -A 50 "QuoteObserver_WithWarmupLateArrivalAndRemoval_MatchesSeriesExactly" "$test_file" | grep -c "\.RemoveAt(" || true)
             has_duplicate=$(grep -A 50 "QuoteObserver_WithWarmupLateArrivalAndRemoval_MatchesSeriesExactly" "$test_file" | grep -c "resend duplicate" || true)
```

Shellcheck (0.11.0)

[warning] has_duplicate appears unused. Verify use (or export if used externally).

(SC2034)